### PR TITLE
More docs: Info messages, missing options, defaults

### PR DESCRIPTION
**Pull request by zuiderkwast**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -224,24 +234,123 @@ Options passed to `start_link/2` as the options `{client_opts, [{connection_opts
...
+    non_neg_integer()`. See options above.
+
+* `#{msg_type := slot_map_updated, slot_map := list(), map_version :=
+  non_neg_integer()}` is sent when the cluster slot-to-node mapping has been
+  updated.
+
+* `#{msg_type := cluster_slots_error_response, response := any()}` is sent when
+  there was an error updating the cluster slot-to-node mapping.
+
+Messages about the connection to a specific node are on the following form:
```
Comment (bjosv) [inline README.md:306]:
```suggestion
Messages about the connection to a specific node are in the following form:
```

Comment (bjosv) [reply README.md:306]: Just a suggestion, felt more natural to me..but not sure

File: README.md [hunk 2]
```diff
@@ -224,24 +234,123 @@ Options passed to `start_link/2` as the options `{client_opts, [{connection_opts
...
+  Reason is as returned by `gen_tcp:connect/4` or `{tls_init, TlsReason}` if
+  `ssl:connect/3` fails.
+
+* `init_error` when one of the initial commands sent to Redis has failed, such
+  as the HELLO command. Reason is a list of error reasons.
+
+* `socket_closed` when the connection has been closed, either by the peer or by
+  ered when an error has happened. Reason is `{recv_exit, RecvReason}` if
+  `gen_tcp:recv/3` or `ssl:recv/3` has failed. RecvReason is typically `timeout`
+  or one of the `inet:posix()` errors. Reason is be `{send_exit, SendReason}` if
```
Comment (bjosv) [inline README.md:339]:
```suggestion
  or one of the `inet:posix()` errors. Reason is `{send_exit, SendReason}` if
```

File: README.md [hunk 3]
```diff
@@ -163,13 +164,18 @@ The following options can be passed to `start_link/2`:
 * `{min_replicas, non_neg_integer()}`
 
   For each Redis master node, the min number of replicas for the cluster
-  to be considered OK.
+  to be considered OK. Default 1.
 
 * `{close_wait, non_neg_integer()}`
 
   How long to delay the closing of clients that are no longer part of
```
[the remark this answers is not in the document]
Comment (zuiderkwast) [reply README.md:171]: It is always milliseconds in Erlang, so it's almost ridiculous to write it everywhere, but sure, I can add it where the type is `non_neg_integer()` (where infinity is not allowed). If the type is `timeout()` it is generally known to be milliseconds.

File: README.md [hunk 4]
```diff
@@ -163,13 +164,18 @@ The following options can be passed to `start_link/2`:
...
   the slot map. The delay is needed so that messages sent to the client
-  are not lost in transit.
+  are not lost in transit. Default 10000.
+
+  It's unlikely that you would get any useful response from a node that is no
+  longer part of the slot map, so you may want to set this option to 0 to make
+  pending commands fail as soon as possible and return an error. This option may
+  be deleted in a future version.
```
Comment (zuiderkwast) [inline README.md:178]: What should we do about this? We need to figure it out.

Comment (zuiderkwast) [reply README.md:178]: What are the messages in transit mentioned?

An idea: The messages in transit might be commands cast from the ered process, so if the ered_client process exits when such cast is in transit, the command gets no response. It should be enough to wait a few milliseconds though, so the command gets queued up in the ered_client process and then it gets an error reply by `reply_all` call in `ered_client:terminate/3`.

Comment (zuiderkwast) [reply README.md:178]: It is `ered_cluster` that stops the `ered_client` anyway when it's been figured out that it's no longer part of the slot map, so after some short time, ered stops sending commands to it.

Comment (zuiderkwast) [reply README.md:178]: Another option is to add some 'stopping' state to `ered_client`, which makes it reject all command immediately, but the process is still alive for some more time to handle messages and perhaps even some response from the socket. Perhaps that's better?

File: README.md (hunk 1 above)
Comment (zuiderkwast) [reply README.md]: I had to look it up. "In the form" seems to be the right phrasing. ("On" is probably my Swedish.)

**Pull request merged**